sdk/storage/azure-storage-common/src/test-shared/java/com/azure/storage/common/test/shared/StorageSpec.groovy

```diff
@@ -19,10 +20,9 @@ class StorageSpec extends Specification {
     private StorageResourceNamer namer
 
     def setup() {
-        def testName = getTestName()
+        def testName = TestHeaderExtension.FeatureInterceptor.getTestName(specificationContext.getCurrentIteration())
```

Comment:
we should create `TestNameProvider` or something like this and extract `getTestName` there and use it in both places.
Can be static code.